### PR TITLE
Refactor validator

### PR DESCRIPTION
**Pull request by louib**
@Ronan- please review.
@xavierhd this is the change to the interface you were expecting.

### CODE REVIEW

File: confirm/validator.py [hunk 1]
```diff
@@ -2,131 +2,146 @@
...
 from confirm.utils import get_most_probable_typo
+from confirm.utils import load_config_file, load_schema_file
 
 
 VALID_TYPES = ('int', 'float', 'bool', 'list', 'str')
 
 
-def validate_config(config, schema, error_on_deprecated=False):
+def load_validation_from_files(schema_file_path, config_file_path):
+    schema = load_schema_file(open(schema_file_path, 'r'))
```
Comment (Ronan-) [inline confirm/validator.py]: So, huh, no error handling at all?

Comment (louib) [inline confirm/validator.py]: We don't do any more error handling than the Yaml library, no.
What would you suggest?

Comment (Ronan-) [inline confirm/validator.py]: Checking whether the files exist and are usable? But yeah, maybe in `load_schema_file()` directly

Comment (Ronan-) [inline confirm/validator.py]: As agreed, we'll leave error handling to the user.

**Comment by Ronan-** [comment]
:+1:

**Pull request merged**